spec/offer-structure.md

```diff
@@ -0,0 +1,133 @@
+  - [4. If a part of a trip is free, a fare with price 0 must be provided.](#4-if-a-part-of-a-trip-is-free-a-fare-with-price-0-must-be-provided)
+  - [5. Pricing is individual (if possible).](#5-pricing-is-individual-if-possible)
+- [Incorrect offer structures](#incorrect-offer-structures)
+
+## Introduction <a name="introduction">
+
+The OSDM offer structure is very powerful and flexible. However if various
+parties interpret the structure in different ways, this can lead to
+interoperability issues. As interoperability is a key goal of OSDM, this
+document defines some rules for the structuring of offers. To ensure real
+interoperability, an offer must not only be syntactically valid, but also comply
+with these rules.
+
+## General rules for Offer parts <a name="Generalrulesofferparts">
+
+The following figure shows the overall structure of an offer in OSDM:
+
+![Offer structure](../images/models/offer-structure.png)
+
+Individual offer mode means that each passenger is given individual admissions
+and reservations. Thus, in individual mode, for every passenger a distinct offer
+part should be provided. In collective mode, one offer part per trip part should
+be provided. Collective mode is used by DB foremost, but is also needed for
+group bookings.
+
+For offer parts such as admissions, reservations, or ancillaries, the following
+rules apply:
+
+### 1. An offer should contain all offer parts per travel class and flexibility/product code.
+
+Assuming first and second class are supported on the given trip, and there exist
+flexible, semi-flexible, and non-flexible products, then the offer should
+contain six offer parts per passenger:
+
+- first class, flexible
+- first class, semi-flexible
+- first class, non-flexible
+- second class, flexible
+- second class, semi-flexible
+- second class, non-flexible
+
+### 2. Per passenger there should be an offer part for different parts of the trip, e.g. legs.
+
+Assuming a trip consists of three legs, then the offer should contain three
+offer parts per passenger, one for each leg of the trip. Or one offer part could
+span multiple legs. Most importantly, there should be no leg without an offer
+part.
+
+### 3. Even if a trip provides partial first class support only, a first class offer is provided.
+
+Assuming a trip consists of three legs, and one leg does not support first
+class. Then the offer should nevertheless contain a first class offer part for
+the whole trip.
+
+### 4. If a part of a trip is free, an offer part with price 0 must be provided.
+
+Again, assuming a trip consists of three legs, and one leg is free of charge.
+Then the offer should nevertheless contain an offer part for that leg, with
+price 0.
+
+## General rules for fares
+
+Apart from offer parts, OSDM also supports the distribution of fares. Fares
```

Comment:
I would explicitly add a rule (or a sentence to the general rules) stating that neither fares nor offer parts may overlap within one offer (As shown in offer-structure-with-fare.png).

For example, it should not be allowed to bundle an offer that contains two fares or offer parts for the same leg but with different flexibilities or classes for a given passenger.